published/external/xdp/details/ioctlfn.h

```diff
@@ -245,6 +245,9 @@ _XdpIoctl(
                 XdpHandle, *Event, NULL, Overlapped, IoStatusBlock, Operation, InBuffer,
                 InBufferSize, OutBuffer, OutputBufferSize));
 
+// Implicit conversion from 'HRESULT' to 'XDP_STATUS'
+#pragma warning(push)
+#pragma warning(disable: 6221)
```

Comment:
```suggestion
#pragma warning(push)
#pragma warning(disable: 6221) // Implicit conversion from 'HRESULT' to 'XDP_STATUS'
```

Reply:
Actually, I wonder if instead of `typedef HRESULT XDP_STATUS` we did `#define XDP_STATUS HRESULT`, would this error go away? Is the problem that this warning requires the type of an HRESULT comparison to equal _exactly_ HRESULT, and does not support typedef'd equivalents?

Reply:
Or, actually, what if we did:

```
#define XDP_STATUS_PENDING                 ((XDP_STATUS)HRESULT_FROM_WIN32(ERROR_IO_PENDING))
```

to ensure each status definition is of type XDP_STATUS rather than having some of subtly different type HRESULT?

Reply:
> Actually, I wonder if instead of `typedef HRESULT XDP_STATUS` we did `#define XDP_STATUS HRESULT`, would this error go away? Is the problem that this warning requires the type of an HRESULT comparison to equal _exactly_ HRESULT, and does not support typedef'd equivalents?

I tried this, but got another many build errors

Reply:
>This cause another build errors.

What build errors did this cause? I expect XDP to build clean with either option.

Reply:
![image](https://github.com/user-attachments/assets/e1e91f54-45bb-4bf2-8b60-104e841533b2)

Reply:
Can you show the changes you made to get that output? It builds cleanly for me when I do `#define XDP_STATUS HRESULT`

Reply:
![image](https://github.com/user-attachments/assets/56707a2a-e39a-46e5-8a5a-720f1ec3bd4b)

This is under `msquic/submodules/xdp-for-windows`.
I built this with msquic

Reply:
Ah, I see, the problem is that `define HRESULT XDP_STATUS` does not define `XDP_STATUS`, it defines `HRESULT`. That's the opposite of what needs to be done per the suggestion above.

Reply:
I'm open to thoughts on this from @nibanks:

1. As a developer depending on several sub-components, it's a _lot_ easier to reason about status codes that are known to be HRESULT/NTSTATUS values as opposed to yet another opaque type. For that reason, my preference is to use the `#define` approach to keep XDP_STATUS values as strictly HRESULTs.
2. As a producer of this API, we have more flexibility if we keep XDP_STATUS as a logically separate type, but it is risky if all values are currently directly convertible to HRESULT/NTSTATUS. It seems like a false abstraction.
3. We could just suppress the error and take no unambiguous stance, which results in maximum confusion.

So as much as I dislike using `#define`, I think in this case it is the ideal solution to do `#define XDP_STATUS HRESULT`.

Reply:
And we should similarly `#define XDP_STATUS NTSTATUS` in kernel mode, for consistency.

Reply:
MsQuic uses `#define` for type definitions, not typedef. I would align with that.

Reply:
> ![image](https://private-user-images.githubusercontent.com/5763034/387374428-56707a2a-e39a-46e5-8a5a-720f1ec3bd4b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIwMzcwMDIsIm5iZiI6MTczMjAzNjcwMiwicGF0aCI6Ii81NzYzMDM0LzM4NzM3NDQyOC01NjcwN2EyYS1lMzlhLTQ2ZTUtOGE1YS03MjBmMWVjM2JkNGIucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTExOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDExMTlUMTcxODIyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9M2RiZWYwMWQxYzE1ZjIwOGE5MDRjYjgyMGI0NTE0NjI2OTI3Nzg1MDc2ZjYyNTRhMDA5YWQ4MTgxMzQxYWQwZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.e1tQubKzku0nHArK75qQuH86keQvlAdkZJqpZCTnXrQ)
> 
> This is under `msquic/submodules/xdp-for-windows`. I built this with msquic

Oops, I did opposite order...